packages/database/src/migrations/20210810040153-FQSVisualsInLESMIS-modifies-data.js

```diff
@@ -0,0 +1,446 @@
+    {
+      transform: 'select',
+      '...': ['label', 'statistic', 'code'],
+    },
+  ],
+});
+
+const DASHBOARD_ITEMS = [
+  {
+    code: 'LESMIS_FQS1_inputs_overview',
+    frontEndConfig: FQS1_PIE_FRONT_END,
+    reportConfig: FQS1_PIE_REPORT,
+  },
+  {
+    code: 'LESMIS_FQS1_inputs',
+    frontEndConfig: FQS1_LIST_FRONT_END,
+    reportConfig: FQS_LIST_REPORT(FQS1_DATA_MAP),
+  },
+  {
+    code: 'LESMIS_FQS2_behaviours_and_processes_overview',
+    frontEndConfig: FQS2_PIE_FRONT_END,
+    reportConfig: FQS2_PIE_REPORT,
+  },
+  {
+    code: 'LESMIS_FQS2_behaviours_and_processes',
+    frontEndConfig: FQS2_LIST_FRONT_END,
+    reportConfig: FQS_LIST_REPORT(FQS2_DATA_MAP),
+  },
+];
+
+// Same util functions as always
+const addNewDashboardItemAndReport = async (
+  db,
+  { code, frontEndConfig, reportConfig, permissionGroup, dashboardCode, entityTypes, projectCodes },
+) => {
+  // insert report
+  const reportId = generateId();
+  const permissionGroupId = (
+    await findSingleRecord(db, 'permission_group', { name: permissionGroup })
+  ).id;
+  await insertObject(db, 'report', {
+    id: reportId,
+    code,
+    config: reportConfig,
+    permission_group_id: permissionGroupId,
+  });
+
+  // insert dashboard item
+  const dashboardItemId = generateId();
+  await insertObject(db, 'dashboard_item', {
+    id: dashboardItemId,
+    code,
+    config: frontEndConfig,
+    report_code: code,
+  });
+
+  // insert relation record connecting dashboard item to dashboard
+  await addItemToDashboard(db, { code, dashboardCode, entityTypes, projectCodes, permissionGroup });
+};
+
+const addItemToDashboard = async (
+  db,
+  { code, dashboardCode, permissionGroup, entityTypes, projectCodes },
+) => {
+  const dashboardItemId = (await findSingleRecord(db, 'dashboard_item', { code })).id;
+  const dashboardId = (await findSingleRecord(db, 'dashboard', { code: dashboardCode })).id;
+  const maxSortOrder = (
+    await findSingleRecordBySql(
+      db,
+      `SELECT max(sort_order) as max_sort_order FROM dashboard_relation WHERE dashboard_id = '${dashboardId}';`,
+    )
+  ).max_sort_order;
+  await insertObject(db, 'dashboard_relation', {
+    id: generateId(),
+    dashboard_id: dashboardId,
+    child_id: dashboardItemId,
+    entity_types: `{${entityTypes.join(', ')}}`,
+    project_codes: `{${projectCodes.join(', ')}}`,
+    permission_groups: `{${permissionGroup}}`,
+    sort_order: maxSortOrder + 1,
+  });
+};
+
+const removeDashboardItemAndReport = async (db, code) => {
+  await db.runSql(`DELETE FROM dashboard_item WHERE code = '${code}';`); // delete cascades to dashboard_relation
+  await db.runSql(`DELETE FROM report WHERE code = '${code}';`);
+};
+
+exports.up = async function (db) {
+  for (const { code, reportConfig, frontEndConfig } of DASHBOARD_ITEMS) {
+    await addNewDashboardItemAndReport(db, {
+      code,
+      reportConfig,
+      frontEndConfig,
+      permissionGroup: 'LESMIS Public',
+      dashboardCode: 'LA_FQS',
```

Comment:
So we already have an FQS dashboard at this point?

Reply:
Yup, it's one of our empty dashboards at the moment